core/src/alpenglow_consensus/voting_loop.rs

```diff
@@ -65,6 +61,57 @@ use {
 /// Banks that have completed replay, but are yet to be voted on
 type PendingBlocks = BTreeMap<Slot, Arc<Bank>>;
 
+/// Context for the block creation loop to start a leader window
+#[derive(Copy, Clone, Debug)]
+pub struct LeaderWindowInfo {
+    pub start_slot: Slot,
+    pub end_slot: Slot,
+    pub parent_block: Block,
+    pub skip_timer: Instant,
+}
+
+/// Communication with the block creation loop to notify leader window
+#[derive(Default)]
+pub struct LeaderWindowNotifier {
+    pub window_info: Mutex<Option<LeaderWindowInfo>>,
+    pub window_notification: Condvar,
+}
+
+// TODO(ashwin): stuffing this here for now will separate once we move to solana-votor
```

Comment:
once I move to new crate i can separate out commitment related stuff into a different file